recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/env python3
+import sys
+import os
+import fcntl
+import struct
+import mmap
+import hashlib
+from optparse import OptionParser
+
+force_update = False
+
+
+def FirmwareUpdateException(str):
+    """Report the error info and exit"""
+    if not force_update:
+        print(str)
+        sys.exit(1)
+
+
+class FirmwareUpdate(object):
+    def __init__(self, force):
+        self.force_update = force
+
+    def sha256_check(self, file):
+        """Firmware Sha256 Check"""
+        check_sha256_file = file+'.sha256'
+        file_sha256 = None
+        try:
+            with open(check_sha256_file, 'r') as f:
+                file_sha256 = str(f.read()).strip()
```

Comment:
Just tested this: Needs to be `str(f.read()).split()[0]` so that it works with regular .sha256 files. Those have the format that sha256sum generates (`<sum> <filename>`).